python/podio_class_generator.py

```diff
@@ -543,6 +535,88 @@ def create_collection(self, classname, definition):
+            lengthName = len(name)
+            klass = member["type"]
+            if(klass in toFormattingStrings  and not klass == "int"):
+                if(formats[klass] > lengthName):
+                    widthOthers = widthOthers + "  int"+ " " + name + "Width = " + toFormattingStrings[klass] + " ; \n"
+                else:
+                    widthOthers = widthOthers + "  int" + " " + name + "Width = " + str(lengthName) + " ; \n"
+            elif klass == "int":
+                findMaximum += "\n    int " + name + "Max ; \n"  
+                findMaximum += "    " + name + "Width = 1 ; \n "
+                findMaximum += "     for(int i = 0 ; i < value.size(); i++){ \n"
+                findMaximum += "         if( value[i].get" + name[:1].upper() + name[1:] + "() > 0 ){ \n" 
+                findMaximum += "            if(" + name + "Max <  value[i].get" + name[:1].upper() + name[1:] + "()){ \n"
+                findMaximum += "               " + name + "Max = value[i].get" + name[:1].upper() + name[1:] + "();"
+                findMaximum += "\n            } \n" 
+                findMaximum += "\n         } \n" 
+                findMaximum += "         else if( -" + name + "Max / 10 > value[i].get" + name[:1].upper() + name[1:] + "()){ \n"
+                findMaximum += "             " + name + "Max = - value[i].get" +  name[:1].upper() + name[1:] + "() * 10; "
+                findMaximum += "\n         } \n"
+                findMaximum += "     } \n"
+                setFormats  += "\n    while(" + name + "Max != 0){ \n"
+                setFormats  += "       " + name + "Width++; \n       " + name + "Max = " + name + "Max / 10; \n    } \n"
+                setFormats  += "   if(" + name + "Width < " + str(lengthName) + "){ " + name + "Width = " + str(lengthName) + ";} \n"
+                WidthIntegers = WidthIntegers + "  " + klass + " " + name + "Width = 1 ; \n"
+            elif(klass == "double" or klass == "float"):
+                if(lengthName > 12):
+                    widthOthers = widthOthers + "  int" + " " + name + "Width = " + str(lengthName) + " ; \n"
+                else:
+                    widthOthers = widthOthers + "  int" + " " + name + "Width = 12 ; \n"
+            elif(klass == "DoubleThree" or klass == "FloatThree"):
+                if(lengthName > 38):
+                    widthOthers = widthOthers + "  int" + " " + name + "Width = " + str(lengthName) + " ; \n"
+                else:
+                    widthOthers += "  int" + " " + name + "Width = 38 ; \n"
+            elif(klass == "IntTwo"):
+                if(lengthName > 24):
+                    widthOthers = widthOthers + "  int" + " " + name + "Width = " + str(lengthName) + " ; \n"
+                else:
+                    widthOthers += "  int" + " " + name + "Width = 24 ; \n"
+            else:
+                widthOthers = widthOthers + "  int" + " " + name + "Width = 38 ; \n"
+            if(klass != "StingVec"):
```

Comment:
StingVec != StringVec ;-)

I do not think hardcoding these types  like IntTwo, DoubleThree, in there is a good idea. Can you think of a better way?